clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -3579,10 +3593,25 @@ void Sema::AddSYCLIntelMaxWorkGroupSizeAttr(Decl *D,
   // a declaration along with [[sycl::reqd_work_group_size(X1, Y1, Z1)]]
   // attribute, check to see if values of reqd_work_group_size arguments are
   // equal or less than values of max_work_group_size attribute arguments.
+  //
+  // The arguments to reqd_work_group_size are ordered based on which index
+  // increments the fastest. In OpenCL, the first argument is the index that
+  // increments the fastest, and in SYCL, the last argument is the index that
+  // increments the fastest.
+  //
+  // The arguments to max_work_group_size are ordered based on which index
+  // increments the fastest. In SYCL, the last argument is the index that
+  // increments the fastest.
   if (const auto *DeclAttr = D->getAttr<ReqdWorkGroupSizeAttr>()) {
-    if (checkWorkGroupSizeAttrValues(DeclAttr->getXDim(), XDim),
-	checkWorkGroupSizeAttrValues(DeclAttr->getYDim(), YDim),
-	checkWorkGroupSizeAttrValues(DeclAttr->getZDim(), ZDim)) {
+    bool checkFirstArgument = getLangOpts().OpenCL
+                              ? checkWorkGroupSizeAttrValues(DeclAttr->getXDim(), XDim)
```

Comment:
Same question here. For openCL - this should be `checkWorkGroupSizeAttrValues(DeclAttr->getXDim(), ZDim) `right?

Reply:
>>For openCL - this should be checkWorkGroupSizeAttrValues(DeclAttr->getXDim(), ZDim) right?

No
The order of the arguments to reqd_work_group_size in SYCL and OpenCL are reversed. In the OpenCL syntax, reqd_work_group_size(X,Y,Z), X is the index that increments the fastest, followed by Y, then Z. If we preserve that definition of X, Y, and Z, then in SYCL the attribute is specified as reqd_work_group_size(Z,Y,X). 
https://github.com/intel/llvm/issues/3252#issuecomment-785261128

Reply:
Fixed now